File: django/config/constants/role.py
from enum import Enum

class UserRole(Enum):
    SUPERADMIN = (1, "Super Admin")
    ADMIN = (2, "Admin")
    USER = (3, "User")

    def __init__(self, code: int, value: str):
        self._code = code
        self._value = value

    @property
    def code(self) -> int:
        return self._code

    @property
    def value(self) -> str:
        return self._value

    @classmethod
    def choices(cls):
        return [(key.code, key.value) for key in cls]

File: react/src/constants/role.js
export class ROLE {
    constructor(code, value) {
        this.code = code;
        this.value = value;
    }

    static SUPERADMIN = new ROLE(1, 'Super Admin');
    static ADMIN = new ROLE(2, 'Admin');
    static USER = new ROLE(3, 'User');
}
